src/store/data-store-level.ts

```diff
@@ -86,21 +134,48 @@ export class DataStoreLevel implements DataStore {
   }
 
   public async has(tenant: string, recordId: string, dataCid: string): Promise<boolean> {
-    const partition = await this.blockstore.partition(tenant);
+    const recordForCid = await this.blockstore.partition(HOST_PARTITION);
+    const tenantForRecord = await recordForCid.partition(dataCid);
+    const dataForTenant = await tenantForRecord.partition(recordId);
+    const tenants = await dataForTenant.partition(tenant);
 
-    const cid = CID.parse(dataCid);
-    const rootBlockBytes = await partition.get(cid);
+    const allowed = await tenants.has(tenant);
+    if (!allowed) {
+      return undefined;
```

Comment:
sanity: `undefined` is not an expected return type. Did you mean to return `false` here? Also consider to just remove this method completely as it appears to be not referenced. If you think we should keep it for whatever reason, add tests for it as it is not tested.